scapy/layers/dot11.py

```diff
@@ -509,7 +509,7 @@ def __init__(self, res=None, name="Dot11List", stats=None):
 
         PacketList.__init__(self, res, name, stats)
     def toEthernet(self):
-        data = map(lambda x:x.getlayer(Dot11), filter(lambda x : x.haslayer(Dot11) and x.type == 2, self.res))
+        data = [x.getlayer(Dot11) for x in filter(lambda x : Dot11 in x and x.type == 2, self.res)]
```

Comment:
Please also replace `filter` (which, by the way, suffers the same Python2 VS Python 3 problem) here, and maybe update the code to
```Python
        data = [x[Dot11] for x in self.res if Dot11 in x and x.type == 2] 
```